src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -410,16 +418,229 @@ def on_request(self, request, **kwargs):
+                                                validate_only=False,
+                                                no_wait=no_wait, handle_extended_json_format=handle_extended_json_format)
+
+
+def validate_arm_template_at_tenant_scope(cmd,
+                                          template_file=None, template_uri=None, parameters=None,
+                                          deployment_name=None, deployment_location=None,
+                                          no_wait=False, handle_extended_json_format=False):
+    return _deploy_arm_template_at_tenant_scope(cli_ctx=cmd.cli_ctx,
+                                                template_file=template_file, template_uri=template_uri, parameters=parameters,
+                                                deployment_name=deployment_name, deployment_location=deployment_location,
+                                                validate_only=True,
+                                                no_wait=no_wait, handle_extended_json_format=handle_extended_json_format)
+
+
+def _deploy_arm_template_at_tenant_scope(cli_ctx,
+                                         template_file=None, template_uri=None, parameters=None,
+                                         deployment_name=None, deployment_location=None,
+                                         validate_only=False,
+                                         no_wait=False, handle_extended_json_format=False):
+    deployment_properties = None
+    if handle_extended_json_format:
+        deployment_properties = _prepare_deployment_properties_unmodified(cli_ctx=cli_ctx, template_file=template_file,
+                                                                          template_uri=template_uri,
+                                                                          parameters=parameters, mode='Incremental')
+    else:
+        deployment_properties = _prepare_deployment_properties(cli_ctx=cli_ctx, template_file=template_file,
+                                                               template_uri=template_uri,
+                                                               parameters=parameters, mode='Incremental')
+
+    mgmt_client = _get_deployment_management_client(cli_ctx, handle_extended_json_format=handle_extended_json_format)
+
+    validation_result = mgmt_client.validate_at_tenant_scope(deployment_name=deployment_name, properties=deployment_properties, location=deployment_location)
+
+    if validation_result and validation_result.error:
+        raise CLIError(validation_result.error)
+    if validate_only:
+        return validation_result
+
+    return sdk_no_wait(no_wait, mgmt_client.create_or_update_at_tenant_scope,
+                       deployment_name, deployment_properties, deployment_location)
+
+
+def _prepare_deployment_properties_unmodified(cli_ctx, template_file=None, template_uri=None, parameters=None, mode=None, rollback_on_error=None):
```

Comment:
>_prepare_deployment_properties_unmodified [](start = 4, length = 41)

there are lots of duplicated code between _prepare_deployment_properties_unmodified and _prepare_deployment_properties, can you refactor these two methods? Besides, I notice unmodified method does not remove operation from the template while modified does not, which is confusing

Reply:
I tried to refactor _prepare_deployment_properties_unmodified and _prepare_deployment_properties, but the current version is the best I could do. If you look at their implementations, there are slight differences. Let me know if you have suggestions on further refactoring. 

As for naming, again it's taken from existing code. I'm not sure if we want to rename the methods. Let me know.